### PR TITLE
Implement  MetadataCombiner, partitial refactoring on Metadata

### PR DESCRIPTION
**Pull request by Wh1isper**
## Motivation and Context

- Rename tests/metadata to tests/data_models
- Intro `RelationshipInspector` for building relations.
- `MetadataCombiner` framework
- Added generic interface to `Metadata`

## Types of changes

- [ ] Maintenance (no change in code, maintain the project's CI, docs, etc.)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Wh1isper** [comment]
Still remaning work:

- [ ] Adding mock data for multiple tables
- [ ] Adding unit tests for `MetadataCombiner`.

File: sdgx/data_models/inspectors/i_id.py
Comment (MooooCat) [inline sdgx/data_models/inspectors/i_id.py:1]: I have a question, what does the new "i" mean in the renamed ID Inspector's path?

Comment (Wh1isper) [reply sdgx/data_models/inspectors/i_id.py:1]: The `id` is a built-in function, and I don't want to have filenames that are the same as the built-in function, which could lead to some confusing problems. The prefix `i` stands for `Inspector`.

Comment (MooooCat) [reply sdgx/data_models/inspectors/i_id.py:1]: Very thoughtful, I'll also notice it in the future.

**Review by MooooCat — APPROVED** [review]
Looks good to me. 

(Maybe we should add more demo data for multi-table scenario, I can start a issue and work on it.)

**Comment by Wh1isper** [comment]
> Maybe we should add more demo data for multi-table scenario, I can start a issue and work on it.

I'll be introducing some very simple mock data to use as unit tests, @MooooCat please help us gather some real multi-table scenario data to introduce into the project, like the current result.csv.

**Pull request merged**